lib/rules/require-unicode-regexp.js

```diff
@@ -44,12 +47,24 @@ module.exports = {
                 const flags = node.regex.flags || "";
 
                 if (!flags.includes("u")) {
-                    context.report({ node, messageId: "requireUFlag" });
+                    context.report({
+                        messageId: "requireUFlag",
+                        node,
+                        suggest: [
+                            {
+                                fix(fixer) {
+                                    return fixer.insertTextAfter(node, "u");
+                                },
+                                messageId: "addUFlag"
+                            }
+                        ]
```

Comment:
We should check if the regex literal would be syntactically valid with the `u` flag, and not provide the suggestion if it wouldn't be valid.

For example:

```js
/\a/
```

If the rule provides a suggestion that adds `u`, after the suggestion is applied the code will have a parsing error:

```js
/\a/u // Parsing error: Invalid regular expression: /\a/: Invalid escape
```

You can check how this is implemented in the [`no-misleading-character-class`](https://github.com/eslint/eslint/blob/1fbf1184fed57df02640aad4659afb54dc26a2e9/lib/rules/no-misleading-character-class.js#L195) rule.

(the same for the RegExp constructor branch)